pkg/minikube/console/console.go

```diff
@@ -0,0 +1,203 @@
+	return Out(format+"\n", a...)
+}
+
+// ErrStyle writes a stylized and formatted error message to stderr
+func ErrStyle(style, format string, a ...interface{}) error {
+	format, err := applyStyle(style, useColor, fmt.Sprintf(format, a...))
+	if err != nil {
+		glog.Errorf("applyStyle(%s): %v", style, err)
+		if oerr := ErrLn(format, a...); oerr != nil {
+			glog.Errorf("Err(%s) failed: %v", format, oerr)
+		}
+		return err
+	}
+	return Err(format)
+}
+
+// Err writes a basic formatted string to stderr
+func Err(format string, a ...interface{}) error {
+	p := message.NewPrinter(preferredLanguage)
+	if errFile == nil {
+		if _, err := p.Fprintf(os.Stderr, "(stderr unset)"+format, a...); err != nil {
+			return err
+		}
+		return fmt.Errorf("no error file has been set")
+	}
+	_, err := p.Fprintf(errFile, format, a...)
+	return err
+}
+
+// ErrLn writes a basic formatted string with a newline to stderr
+func ErrLn(format string, a ...interface{}) error {
+	return Err(format+"\n", a...)
+}
+
+// Success is a shortcut for writing a styled success message to stdout
+func Success(format string, a ...interface{}) error {
+	return OutStyle("success", format, a...)
+}
+
+// Fatal is a shortcut for writing a styled fatal message to stderr
+func Fatal(format string, a ...interface{}) error {
+	return ErrStyle("fatal", format, a...)
+}
+
+// Warning is a shortcut for writing a styled warning message to stderr
+func Warning(format string, a ...interface{}) error {
+	return ErrStyle("warning", format, a...)
+}
+
+// Failure is a shortcut for writing a styled failure message to stderr
+func Failure(format string, a ...interface{}) error {
+	return ErrStyle("failure", format, a...)
+}
+
+// SetPreferredLanguageTag configures which language future messages should use.
+func SetPreferredLanguageTag(l language.Tag) {
+	glog.Infof("Setting Language to %s ...", l)
+	preferredLanguage = l
+}
+
+// SetPreferredLanguage configures which language future messages should use, based on a LANG string.
+func SetPreferredLanguage(s string) error {
+	if s == "" || s == "C" {
+		SetPreferredLanguageTag(defaultLanguage)
+		return nil
+	}
+	// Ignore encoding preferences: we always output utf8. Handles "de_DE.utf8"
+	parts := strings.Split(s, ".")
+	l, err := language.Parse(parts[0])
+	if err != nil {
+		return err
+	}
+	SetPreferredLanguageTag(l)
+	return nil
+}
+
+// SetOutFile configures which writer standard output goes to.
+func SetOutFile(w fdWriter) {
+	glog.Infof("Setting OutFile to fd %d ...", w.Fd())
+	outFile = w
+	useColor = wantsColor(w.Fd())
+}
+
+// SetErrFile configures which writer error output goes to.
+func SetErrFile(w fdWriter) {
+	glog.Infof("Setting ErrFile to fd %d...", w.Fd())
+	errFile = w
+	useColor = wantsColor(w.Fd())
+}
+
+// wantsColor determines if the user might want colorized output.
+func wantsColor(fd uintptr) bool {
+	glog.Infof("%s=%q\n", OverrideEnv, os.Getenv(OverrideEnv))
+	switch os.Getenv(OverrideEnv) {
+	case "0":
```

Comment:
I would consider using "true/false" or "on/off" or "yes/no" over "0/1"

Reply:
Normally you set the env var to enable, and leave it unset to disable.

In Go this means "" and "1", unless you want to bother with LookupEnv ?

Reply:
Since the idea here is to enable the user to opt _out_ of this, and enable it by default...
Maybe we should change the variable name altogether ? like MINIKUBE_NOEMOJIS

Or keep the current approach, and use a string (not "bool") - and let "" mean auto-determine.
Just have to add some more case statements, to cover the above alternative spellings.

Reply:
I've switched this code to use strconv.ParseBool to allow true/false as well as 0/1. NOTE: Other parts of minikube use the ladder.

I've added some comments as well, to clarify the behavior. I want users to also be able to force color on, if our automatic sniffing isn't appropriate for their environment.